include/deal.II/lac/dynamic_sparsity_pattern.h

```diff
@@ -446,6 +446,15 @@ class DynamicSparsityPattern : public Subscriptor
   bool
   exists(const size_type i, const size_type j) const;
 
+  /**
+   * Get a view of this sparsity pattern and store the result in @p view.
+   * That is, for all rows in @p rows extract non-empty columns.
+   * The resulting sparsity pattern will have number of rows equal
+   * `rows.n_elements()`.
+   */
+  void
+  get_view(DynamicSparsityPattern &view, const IndexSet &rows) const;
```

Comment:
Don't you think that creating the view the other way around, i.e., letting the calling sparsity pattern hold the result of the view from a more comprehensive sparsity pattern given as input, would be preferable? Since this is what `IndexSet::get_view` does, see
https://www.dealii.org/current/doxygen/deal.II/classIndexSet.html#a9005c00d85ddaa6598081148a148c7c0
I would suggest we do the same here.

Reply:
agree, will change.

Reply:
@kronbichler 

`IndexSet::get_view()` returns the result (a view) based on the caller object. 

But I am fine changing the interface nonetheless (recompiling at the moment...)

Reply:
Right, that is slightly different - I guess it depends on whether we want to return a big object like a sparsity pattern (how would that fit into your application) by value. As RTO will get used, I think that would not be too bad. I guess it depends on what you see fit and what others think. We might choose a different name `construct_by_view` or something like that if we keep the convention of letting the caller get modified.

Reply:
> whether we want to return a big object like a sparsity pattern

but we can't return by value as `operator=` is valid only for empty objects.

Reply:
apparently RTO/RVO works 👍

Reply:
The `return` statement should call the move constructor on the returned object. But it may be elided, which I suspect is happening here, with the result constructed right in its final location.